pages/library/[id].js

```diff
@@ -0,0 +1,149 @@
+import { useRouter } from "next/router";
+import styled from "styled-components";
+import { useEffect } from "react";
+import { useRef } from "react";
+import Image from "next/image";
+import Link from "next/link";
+
+const StyledForm = styled.form`
+  display: grid;
+  gap: 0.2rem;
+  flex-direction: column;
+  grid-template-columns: 50% 50%;
+  margin-top: 16rem;
+  color: black;
+  justify-content: center;
+  align-items: center;
+  font-family: Georgia, "Times New Roman", Times, serif;
+`;
+
+const StyledLabel = styled.label`
+  text-align: right;
+`;
+
+const StyledInput = styled.textarea`
+  background-color: transparent;
+`;
+
+const StyledButton = styled.button`
+  display: flex;
+  position: absolute;
+  margin-top: 2rem;
+  top: 23rem;
+  text-align: center;
+  justify-self: center;
+  justify-self: center;
+  font-family: Georgia, "Times New Roman", Times, serif;
+  font-weight: bold;
+`;
+
+const StyledSelect = styled.select`
+  background-color: transparent;
+`;
+
+const StyledBook = styled(Image)`
+  position: absolute;
+  top: 0;
+  left: 0;
+  object-fit: contain;
+  width: 100%;
+  height: 100%;
+  z-index: -2;
+`;
+
+const StyledBackToBookImage = styled(Image)`
+  position: relative;
+  margin-top: 10rem;
+`;
+
+const BackgroundImage = styled(Image)`
+  position: absolute;
+  top: 0;
+  left: 0;
+  opacity: 0.7;
+  z-index: -3;
+  object-fit: cover;
+`;
+
+export default function EditWisdom({
+  library,
+  handleEditWisdomSubmit,
+  currentBook,
+}) {
+  const inputRef = useRef(null);
+
+  function handleSubmit(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const wisdomData = Object.fromEntries(formData);
+    handleEditWisdomSubmit({
+      ...wisdomData,
+      answeredRight: 0,
+      id: wisdom.id,
+      book: currentBook,
+    });
+    router.push("/library/viewBook");
+  }
+  useEffect(() => {
+    inputRef.current.focus();
+  }, []);
+  const router = useRouter();
+  const { id } = router.query;
```

Comment:
A small recommendation:
By declaring const router = useRouter(); and const { id } = router.query; at the beginning of the component, you also improve code readability and avoid unnecessary repetitions of code when accessing the route object and dynamic route parameter at different parts of your component.